easybuild/tools/toolchain/toolchain.py

```diff
@@ -415,9 +453,20 @@ def _add_dependency_variables(self, names=None, cpp=None, ld=None):
         else:
             deps = [{'name': name} for name in names if name is not None]
 
-        for root in self.get_software_root([dep['name'] for dep in deps]):
-            self.variables.append_subdirs("CPPFLAGS", root, subdirs=cpp_paths)
-            self.variables.append_subdirs("LDFLAGS", root, subdirs=ld_paths)
+        dep_names = []
+        for dep in deps:
+            # name may be None or a list of names for external modules
```

Comment:
then check that, all other modules should raise error (and stop doing magic/hackish things like dynamic types)